go/vt/vtgate/executor.go

```diff
@@ -59,7 +59,7 @@ var (
 	errNoKeyspace     = vterrors.Errorf(vtrpcpb.Code_FAILED_PRECONDITION, "no keyspace in database name specified. Supported database name format (items in <> are optional): keyspace<:shard><@type> or keyspace<[range]><@type>")
 	defaultTabletType topodatapb.TabletType
 
-	queriesProcessed = stats.NewCountersWithSingleLabel("QueriesProcessed", "Queries processed at vtgate by plan type", "Plan")
+	queriesProcessed = stats.NewCountersWithMultiLabels("QueriesProcessed", "Queries processed at vtgate by plan type, keyspace and table", []string{"Plan", "Keyspace", "Table"})
```

Comment:
Adding labels to an existing metric can confuse downstream systems, including Prometheus. This isn't necessarily the end of the world, but overall I think it would be cleaner to use a new metric name if we're going to change the labels.

Maybe we rename this to simply "Queries"?

I also think we should extend `QueriesRouted` to include the same set of labels since it's nice to keep those two in sync to measure the percentage of scatter queries that are happening.

I'd then rename that one to "ShardQueries"

Reply:
I did this. However one thing... We are going to break dashboards of people that relied on the old metric name. 

Wondering if we should also emit the old one or too overkill. 

@demmer  do you have thoughts ?

Reply:
Actually, I rolled back the name changes. It breaks the tests: https://travis-ci.org/vitessio/vitess/jobs/563803123#L572

This is due to:
```
panic: Reuse of exported var name: Queries
```

The existent one comes from tabletenv: 
```
	QueryStats = stats.NewTimings("Queries", "MySQL query timings", "plan_type")
```

The problem happens in vtcombo. 

I think we should keep the old names.

Reply:
To be honest I'm not really sure what the best path forward is here...

My understanding is that Prometheus' data model expects that the labels used for a given metric shouldn't change, since the label + name is used as the underlying key in the timeseries database. So by changing the labels you're effectively creating a new metric and so during a deploy things might be tricky.

To some degree the right thing is to keep the old name and labels exactly as-is, but then add a new metric of something like `QueriesByTable` and for below `QueriesRoutedByTable`.

That way during a deploy existing metrics stay exactly the same but then we would eventually deprecate them.

However this seems like a lot of overkill just for metrics.